docs/content/development/extensions-core/druid-lookups.md

```diff
@@ -0,0 +1,122 @@
+# Cached Lookup Module
+
+## Description
+This module provides a one to one caching mechanism for a specific lookup table source like RDBMS or distributed file systems.
+The main goal of this cache is to speed up the access to a high latency lookup sources and to provide a caching isolation for every lookup source.
+Thus user can define various caching strategies or and implementation per lookup, even if the source is the same.
+
+## Architecture
+Generally speaking this module can be divided into two main component, namely, the data fetcher layer and caching layer.
+
+### Data Fetcher layer:
+
+First part is the data fetcher layer API `DataFetcher`, that exposes a set of fetch methods to fetch data from the actual Lookup dimension source.
+For instance `JdbcDataFetcher` provides an implementation of `DataFetcher` that can be used to fetch key/value from a RDBMS via JDBC driver.
+If you need new type of data fetcher, all you need to do, is to implement the interface `DataFetcher` and load it via another druid module.
+### Caching layer
+
+This extension comes with two different caching strategies. First strategy is a poll based and the second is a load based.
+#### Poll lookup cache:
```

Comment:
if it is not a big deal i prefer to keep poll, since all the classes naming reflect that, plus it is pretty common name when you read about caching in general.